osgar/drivers/usbcam.py

```diff
@@ -0,0 +1,47 @@
+"""
+  Publish jpg images captured using libuvc (pupil-labs fork).
+  Only cameras directly supporting jpg output are supported.
+"""
+
+import uvc
+from threading import Thread
+
+from osgar.bus import BusShutdownException
+
+
+class UsbCam:
+    def __init__(self, config, bus):
+        self.input_thread = Thread(target=self.run_input, daemon=True)
+        self.bus = bus
+
+        bus = config.get('bus', 1)
+        port = config.get('port', 0)
+        self.sleep = config.get('sleep', 10)
+        
+        dev_list = uvc.device_list()
+        for dev in dev_list:
+            if dev['bus_number'] == bus and dev['bus_ports'][0] == port:
+                self.cap = uvc.Capture(dev["uid"])
+                self.cap.frame_mode = self.cap.avaible_modes[0]
+                self.cap.bandwidth_factor = 1.2
+                return
+
+        print("camera not found")
```

Comment:
as discussed over mail - I would not put `sys.exit()` here but at least logged into log via `self.bus.report_error()`. Maybe we need two phase initialization before `start()`, i.e. that all modules were properly initialized??

Reply:
ok